src/v/cluster/types.h

```diff
@@ -987,6 +987,57 @@ struct abort_group_tx_reply
     auto serde_fields() { return std::tie(ec); }
 };
 
+struct find_coordinator_request
+  : serde::envelope<
+      find_coordinator_request,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    kafka::transactional_id tid;
+
+    find_coordinator_request() noexcept = default;
+
+    find_coordinator_request(kafka::transactional_id tid)
+      : tid(std::move(tid)) {}
+
+    friend bool
+    operator==(const find_coordinator_request&, const find_coordinator_request&)
+      = default;
+
+    friend std::ostream&
+    operator<<(std::ostream& o, const find_coordinator_request& r);
+
+    auto serde_fields() { return std::tie(tid); }
+};
+
+struct find_coordinator_reply
+  : serde::envelope<
+      find_coordinator_reply,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    std::optional<model::node_id> coordinator{std::nullopt};
```

Comment:
we anyway need to send address back to the client so there is still a chance of change. also it's better to turn ntp into an address (node id) at a central location to avoid different nodes deciding differently